File: backend/catalog/src/main/java/com/fluytcloud/catalog/entities/Template.java
package com.fluytcloud.catalog.entities;

import com.fluytcloud.scm.entities.Scm;

public record Template(
        Integer id,
        String name,
        Scm templateScm,
        String templateRepository,
        String templateBranch,
        Scm targetScm,
        String targetBranch
) {
}

File: frontend/src/app/system/catalog/template/template.ts
import {Scm} from "../../settings/scm/scm";

export interface Template {
  id: number,
  name: string,
  templateScm: Scm,
  templateRepository: string,
  templateBranch: string,
  targetScm: Scm,
  targetBranch: string
}
